snooty/test_parser.py

```diff
@@ -244,6 +244,63 @@ def test_admonition() -> None:
     )
 
 
+def test_toml_replacement() -> None:
+    path = ROOT_PATH.joinpath(Path("test.rst"))
+    project_config = ProjectConfig(
+        ROOT_PATH,
+        "",
+        source="./",
+        substitutions={"yaml": "*Yet Another Markup Language*"},
+    )
+    parser = rstparser.Parser(project_config, JSONVisitor)
+    substitution_nodes = {}
+    for k, v in project_config.substitutions.items():
+        node, diagnostics = parse_rst(parser, ROOT_PATH, v)
+        substitution_nodes[k] = node.ast
+
+    project_config.substitution_nodes = substitution_nodes
+    parser = rstparser.Parser(project_config, JSONVisitor)
+
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
+|yaml| ain't markup language.
+""",
+    )
+    assert diagnostics == []
+    check_ast_testing_string(
+        page.ast,
+        """<root>
+        <paragraph>
+        <substitution_reference>
```

Comment:
The `<substitution_reference>` should be gone too, shouldn't it?

Reply:
At the moment, we're taking the child nodes returned by `parse_rst()` and assigning them to the `children` property of the substitution reference node. With this approach, I believe the `<substitution_reference>` tag should still be present. I also think that keeping it in the AST can help with debugging on the front-end. But if you feel that it isn't logical to include it in the AST, I'm open to that approach as well!